### PR TITLE
Sql split core

### PR DESCRIPTION
**Pull request by askarovit**

### CODE REVIEW

File: core/server.ts [hunk 1]
```diff
@@ -10,11 +10,11 @@ app.use(bodyParser.json());
 const loader = new Loader();
 loader.initRoute(app);
 loader.runMigrate(db)
-    .then(res => {
+    .then(() => {
         /**
          * TODO Change on logger
          */
-        console.log('*** Migrate Done ****');
+        console.log('Migrate Done.');
```
Comment (alex-arugon) [inline core/server.ts]: Strange name, maybe `migration` is better. Also `console.log` should be replaced with logger

File: test/shared/migrate/runMigrate.test.ts [hunk 2]
```diff
@@ -16,17 +16,51 @@ describe('Migrate on create tables', () => {
     });
 
     after( (done) => {
-        const filePath = path.join(pathMockData, 'dropTables.sql');
-        const sql = new QueryFile(filePath, { minify: true });
+        const sql = getSQLRequest('dropTables.sql');
         db.query(sql)
             .then(() => {
```
Comment (alex-arugon) [inline test/shared/migrate/runMigrate.test.ts]: Don't use done() its redundant, but use await + try/catch here

File: test/shared/migrate/runMigrate.test.ts [hunk 3]
```diff
@@ -16,17 +16,51 @@ describe('Migrate on create tables', () => {
...
     });
 
-    it('It should create tables', async () => {
-        await migrate.run();
+    it('It should create tables', (done) => {
+        migrate.run()
+            .then(() => {
+                const sql = getSQLRequest('getTables.sql');
+                db.query(sql)
+                    .then((result) => {
```
Comment (alex-arugon) [inline test/shared/migrate/runMigrate.test.ts]: try { async/await } catch () {}

File: test/shared/migrate/runMigrate.test.ts [hunk 4]
```diff
@@ -16,17 +16,51 @@ describe('Migrate on create tables', () => {
...
+                    .catch((err) => {
+                        done(err);
+                    });
+            });
+    });
 
+    it('It should to exist tables', (done) => {
+        const sql = getSQLRequest('isExistTable.sql');
+        db.query(sql)
+            .then((result) => {
```
Comment (alex-arugon) [inline test/shared/migrate/runMigrate.test.ts]: try { async/await } catch () {}

File: core/database/migrations/20160723182900_createSchema.sql [hunk 5]
```diff
@@ -100,6 +100,14 @@ CREATE TABLE IF NOT EXISTS "forks_stat" (
   "cause"             INT         NOT NULL
 );
 
+CREATE TABLE IF NOT EXISTS "multisignatures" (
```
Comment (alex-arugon) [inline core/database/migrations/20160723182900_createSchema.sql]: I wonder if we should remove all functionality, corresponding to multisignatures?

File: core/database/migrator.ts [hunk 6]
```diff
@@ -1,55 +1,85 @@
...
-
-    read(): Array<Object> {
-        const migrationsPath = path.join(process.cwd(), 'core/database', 'migrations');
-        let fileMigration: Array<IPropertyFile> = [];
+    constructor(db: IDatabase<any>) {
+        this.db = db;
+    }
 
-        fs.readdirSync(migrationsPath).forEach(file => {
+    async run(migrationsPath ?: string) {
```
Comment (alex-arugon) [inline core/database/migrator.ts]: why does this function return value only in `if` section?

File: core/database/migrator.ts [hunk 7]
```diff
@@ -1,55 +1,85 @@
...
-                     })
-                     .catch(err => {
-                         return new Response({
-                             errors: err.message
-                         })
-                     });
-             })
-        });
+        for (let file of files) {
+            console.log(file.name);
```
Comment (alex-arugon) [inline core/database/migrator.ts]: ?

File: test/shared/migrate/runMigrate.test.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import chai, { expect } from 'chai';
+import { IDatabase, QueryFile } from 'pg-promise';
+import { DatabaseConnector } from './mock/db';
+import { Migrator } from 'core/database/migrator';
+
+describe('Migrate on create tables', () => {
+    let migrate: Migrator;
+    let pathSQL;
+
+    before(async (done) => {
```
Comment (alex-arugon) [inline test/shared/migrate/runMigrate.test.ts]: why this function is described as async, but there is no await in it?

File: core/database/migrations/20160723182900_createSchema.sql [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+  "timestamp"            INT    NOT NULL,
+  "height"               INT    NOT NULL,
+  "previousBlock"        CHAR(64),
+  "numberOfTransactions" INT    NOT NULL,
+  "totalAmount"          BIGINT NOT NULL,
+  "totalFee"             BIGINT NOT NULL,
+  "reward"               BIGINT NOT NULL,
+  "payloadLength"        INT    NOT NULL,
+  "payloadHash"          BYTEA  NOT NULL,
+  "generatorPublicKey"   BYTEA  NOT NULL,
```
Comment (alex-arugon) [inline core/database/migrations/20160723182900_createSchema.sql]: Do we need to store all such fields in `hex` format?

File: core/database/migrator.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Read file migration and run it
+     * @param files
+     */
+    async runMigrate(files: Array<any>) {
+        for (let file of files) {
+            const sql = new QueryFile(file.path, { minify: true });
+
+            await this.db.query(sql)
```
Comment (alex-arugon) [inline core/database/migrator.ts]: await and then? use new async await notation

File: core/database/migrations/20160723182900_createSchema.sql (hunk 9 above)
Comment (askarovit) [reply core/database/migrations/20160723182900_createSchema.sql]: I don't know

**Pull request merged**